tools/fw-update/rs-fw-update.cpp

```diff
@@ -546,6 +546,7 @@ try
 
     std::cout << std::endl << "Waiting for device to reconnect..." << std::endl;
     std::unique_lock<std::mutex> lk(mutex);
+    new_device = rs2::device();  // otherwise the wait will exit right away
```

Comment:
On which device did you see it?
I am not sure that's the case on all devices/platforms

How is 
rs2::device new device;

different from 
rs2::device new device = rs2::device() ;
?

Reply:
`new_device` is updated every time the callback sees a new device. If the fw-update saw a new device (which is likely, and does happen with D555), then it will be set and the wait will not wait.

I tested with D455 and D555.

Reply:
It waited for a recovery device:
` if (d.is<rs2::update_device>...`

IMO your new code now can override the device the callback already set and we will not get a notification of a new device.
Just wait until timeout right?

Maybe DDS should be handled differently?

Reply:
We just restarted the device... why does it not make sense to reset the device we're waiting on? Has nothing to do with DDS.
I tested on D455... it works

Reply:
`I tested on D455... it works`
On your system, with current load, but it's not robust.

Add this line and you will see you will wait 15 seconds for timeout.
![image](https://github.com/user-attachments/assets/803e7493-89ed-40b9-a8ec-9ea73c767234)

The original solution does work.
On D500 when it takes longer to enumerate we do wait.
If you wish to reset the 'new_device'  you wait on (to solve DDS related issue) try doing it 1 line before the `update` call which cause a device reset when the update end

Reply:
I think equating a 3-second wait with a few lines of code while a device is rebooting is stretching it, but ok:
I'll move the reset to above line 530, before we set `done = true`, ok?

Reply:
Why not before the update?
Is it in use in this function?

Reply:
Moved before the update and tested.